fcntl.flock no longer typechecks with io.FileIO argument

Seems like an edge case broken by the change in https://github.com/python/typeshed/pull/1505:

This code works in reality, but mypy does not like it:

```python
import io
import fcntl


with io.FileIO('hello.txt', mode='a+', closefd=True) as f:
    fcntl.flock(f, fcntl.LOCK_EX)
    print("hey it works")
    fcntl.flock(f, fcntl.LOCK_UN)
```

```
$ python ./blurp.py
hey it works
$ mypy ./blurp.py
blurp.py:6: error: Argument 1 to "flock" has incompatible type "IOBase"; expected "Union[int, IO[Any]]"
blurp.py:8: error: Argument 1 to "flock" has incompatible type "IOBase"; expected "Union[int, IO[Any]]"
```

--- a/stdlib/3/fcntl.pyi
+++ b/stdlib/3/fcntl.pyi
@@ -1,4 +1,5 @@
 # Stubs for fcntl
+from io import IOBase as _IOBase
 from typing import Any, IO, Union
 
 FASYNC = ...  # type: int
@@ -74,7 +75,7 @@ LOCK_SH = ...  # type: int
 LOCK_UN = ...  # type: int
 LOCK_WRITE = ...  # type: int
 
-_AnyFile = Union[int, IO[Any]]
+_AnyFile = Union[int, IO[Any], _IOBase]
 
 # TODO All these return either int or bytes depending on the value of
 # cmd (not on the type of arg).

--- a/stdlib/3/fcntl.pyi
+++ b/stdlib/3/fcntl.pyi
@@ -1,5 +1,5 @@
 # Stubs for fcntl
-from io import IOBase as _IOBase
+from io import IOBase
 from typing import Any, IO, Union
 
 FASYNC = ...  # type: int
@@ -75,7 +75,7 @@ LOCK_SH = ...  # type: int
 LOCK_UN = ...  # type: int
 LOCK_WRITE = ...  # type: int
 
-_AnyFile = Union[int, IO[Any], _IOBase]
+_AnyFile = Union[int, IO[Any], IOBase]
 
 # TODO All these return either int or bytes depending on the value of
 # cmd (not on the type of arg).

--- a/stdlib/3/fcntl.pyi
+++ b/stdlib/3/fcntl.pyi
@@ -1,6 +1,5 @@
 # Stubs for fcntl
 from typing import Any, IO, Union
-import typing
 
 FASYNC = ...  # type: int
 FD_CLOEXEC = ...  # type: int

